pyomo/contrib/mindtpy/algorithm_base_class.py

```diff
@@ -1626,7 +1683,11 @@ def solve_main(self):
                         "No-good cuts are added and GOA algorithm doesn't converge within the time limit. "
                         'No integer solution is found, so the CPLEX solver will report an error status. '
                     )
-            return None, None
+            # Value error will be raised if the MIP problem is unbounded and appsi solver is used when loading solutions. Although the problem is unbounded, a valid result is provided and we do not return None to let the algorithm continue.
+            if 'main_mip_results' in dir():
```

Comment:
`dir()` returns a list, so this `if` statement is linear time, which isn't ideal. Is there a more efficient way to check that you have MIP results?

Reply:
Is it better to use `locals()`, which returns a dict?

Reply:
If you're going to use `in` then a dict would be much better, yes.